frontend/readline.c

```diff
@@ -8,6 +8,9 @@
 #include <string.h>
 #include <signal.h>
 #include <unistd.h>
+#include <libgen.h>
```

Comment:
```suggestion

#undef __generic
#include <libgen.h>

```
Something like this ought to do it.